src/plist.h

```diff
@@ -314,35 +314,46 @@ EXPORT_INLINE Obj NewImmutableEmptyPlist(void)
 /****************************************************************************
 **
 *F  NewPlistFromArray(<list>,<length>) . . create a plain list from a C array
+*F  NewPlistFromArgs(<args...>) .  create a plain list from list of arguments
+**
+**  NewPlistFromArgs turns a variable-length list of macro arguments into
+**  an array, which is then passed to NewPlistFromArray.
+**
+**  There is no (nice) single implementation of NewPlistFromArgs that works in
+**  both C and C++, so there are two seperate implementations.
 */
+
 EXPORT_INLINE Obj NewPlistFromArray(const Obj * list, Int length)
 {
     if (length == 0) {
         return NewEmptyPlist();
     }
-
     Obj o = NEW_PLIST(T_PLIST, length);
     SET_LEN_PLIST(o, length);
     memcpy(BASE_PTR_PLIST(o), list, length * sizeof(Obj));
     CHANGED_BAG(o);
     return o;
 }
 
-/****************************************************************************
-**
-*F  NewPlistFromArgs(<args...>) .  create a plain list from list of arguments
-**
-**  This macro turns a variable-length list of macro arguments into an array,
-**  which is then passed to NewPlistFromArray.
-**
-**  __VA_ARGS__ contains the list of arguments given to this macro.
-**  (Obj[]){ __VA_ARGS__ } creates an array of Obj containing the elements
-**  of __VA_ARGS__.
-*/
+#ifdef __cplusplus
+ // escape extern "C"
+
+extern "C++"
+template <unsigned int length>
+EXPORT_INLINE Obj NewPlistFromArrayCppWrap(const Obj (&list)[length])
+{
+    return NewPlistFromArray(list, length);
+}
+
+#define NewPlistFromArgs(...)                                                \
+    NewPlistFromArrayCppWrap({ __VA_ARGS__ })
```

Comment:
I am afraid this requires C++11:
```
src/pperm.cc: In function ‘UInt** FuncINDEX_PERIOD_PPERM(Obj, Obj)’:
src/plist.h:347:30: error: extended initializer lists only available with -std=c++11 or -std=gnu++11 [-Werror]
     NewPlistFromArrayCppWrap({ __VA_ARGS__ })
                              ^
```

But if we use C++11, we might as well use a nicer implementation using variadic templates:
```C++
template<typename... Ts>
Obj NewPlistFromArgs(Ts... args)
{
    const int size = sizeof...(args);
    Obj res[size] = {args...};
    return NewPlistFromArray(res, size);
}
```

Reply:
Woops, I should have checked if this worked with C++03 explictly!

Reply:
Also, we'd then have to decide that we really want to require C++11. Might then also require C11. Both standards are 8 years old... I am still not sure whether we could convince everybody? I guess somewhat older compilers supporting partial (pre-)version fo C++11 would also be OK? According to  <https://en.cppreference.com/w/cpp/compiler_support>, variadic templates arrived in GCC 4.4 (first released April 2009) resp. 4.3 (first released March 2008, but only supports an earlier version of the spec, which may or may not be sufficient for our needs).

Of course initializer lists also were introduced in GCC 4.4 (cf same website). I am not sure what "extended initializer lists" refers to, though. 

In any case, we'd have to use something like https://www.gnu.org/software/autoconf-archive/ax_cxx_compile_stdcxx_11.html(possibly tweaked to accept `-std=gnu++1x` or so?)

Reply:
I guess https://www.gnu.org/software/autoconf-archive/ax_cxx_compile_stdcxx.html is the macro to use -- and looking at it, it seems to accept `c++0x` as a fallback, so perhaps we'd be OK with that -- using a 10 years old standard ought to be fine. Right? Right?

Reply:
I'd think so -- I've been using C++0x in some packages and other programs, and haven't had any complains for a few years.

Reply:
I know that @hulpke sometimes is using a cluster which "only" has gcc version 4.4.7 20120313, but that one has perfectly fine support for either implementation of this PR, so should work fine.

Reply:
Also, the machine giot finally updated, so this gcc issue seems to be gone. But thanks for thinking of me!